cloudwatch/src/lambda_function.py

```diff
@@ -77,34 +71,52 @@ def _parse_cloudwatch_log(log, aws_logs_data, log_type):
         pass
 
 
-def _enrich_logs_data(aws_logs_data, context):
-    # type: (dict, 'LambdaContext') -> None
+def _get_additional_logs_data(aws_logs_data, context):
+    # type: (dict, 'LambdaContext') -> dict
+    additional_fields = ['logGroup', 'logStream', 'messageType', 'owner']
+    additional_data = dict((key, aws_logs_data[key]) for key in additional_fields)
+    try:
+        additional_data['function_version'] = context.function_version
+        additional_data['invoked_function_arn'] = context.invoked_function_arn
+    except KeyError:
+        pass
+
     try:
-        aws_logs_data['function_version'] = context.function_version
-        aws_logs_data['invoked_function_arn'] = context.invoked_function_arn
+        # If ENRICH has value, add the properties
+        if os.environ['ENRICH']:
+            properties_to_enrich = os.environ['ENRICH'].split(";")
+            for property_to_enrich in properties_to_enrich:
+                property_key_value = property_to_enrich.split("=")
+                additional_data[property_key_value[0]] = property_key_value[1]
     except KeyError:
         pass
 
+    try:
+        additional_data['type'] = os.environ['TYPE']
+    except KeyError:
+        additional_data['type'] = 'logzio_cloudwatch_lambda'
```

Comment:
Is it possible to get this configuration from yaml? also, you could print to the user that no TYPE variable was found, since he might not know it is case sensitive and used, for example, "type" instead

Reply:
The common practice in lambda is to use env var